kyo-offheap/native/src/main/java/lang/foreign/MemorySegment.scala

```diff
@@ -0,0 +1,78 @@
+package java.lang.foreign
+
+import scala.scalanative.unsafe._
+import scala.scalanative.libc.string.memcpy
+import scala.scalanative.unsigned._
+
+/** A minimal stub for Java's MemorySegment using Scala Native pointers.
+  *
+  * This implementation wraps a pointer along with the allocated size in bytes.
+  * It also provides basic support for slicing and for reading/writing primitive values,
+  * which are used by the Layout instances in the shared code.
+  */
+final class MemorySegment(val ptr: Ptr[Byte], val byteSize: Long) {
+  /** Creates a new MemorySegment that is a slice of this segment.
+    *
+    * @param offset The offset (in bytes) at which the slice begins.
+    * @param newSize The size (in bytes) of the new segment.
+    * @return A new MemorySegment representing the slice.
+    */
+  def asSlice(offset: Long, newSize: Long): MemorySegment = {
+    if (offset < 0 || newSize < 0 || offset + newSize > byteSize)
+      throw new IllegalArgumentException("Invalid slice parameters")
+    new MemorySegment(ptr + offset, newSize)
+  }
+
+  /** Reads a value from memory using the provided layout.
+    */
+  def get(layout: Any, offset: Long): Any =
+    if (layout == ValueLayout.JAVA_BYTE) {
+      !(ptr + offset).cast[Byte]
+    } else if (layout == ValueLayout.JAVA_SHORT) {
+      !(ptr + offset).cast[Short]
+    } else if (layout == ValueLayout.JAVA_INT) {
+      !(ptr + offset).cast[Int]
+    } else if (layout == ValueLayout.JAVA_LONG) {
+      !(ptr + offset).cast[Long]
+    } else if (layout == ValueLayout.JAVA_FLOAT) {
+      !(ptr + offset).cast[Float]
+    } else if (layout == ValueLayout.JAVA_DOUBLE) {
+      !(ptr + offset).cast[Double]
+    } else {
+      throw new UnsupportedOperationException("Unsupported layout in get")
+    }
+
+  /** Writes a value to memory using the provided layout.
+    */
+  def set(layout: Any, offset: Long, value: Any): Unit =
+    if (layout == ValueLayout.JAVA_BYTE) {
+      !(ptr + offset).cast[Byte] = value.asInstanceOf[Byte]
+    } else if (layout == ValueLayout.JAVA_SHORT) {
+      !(ptr + offset).cast[Short] = value.asInstanceOf[Short]
+    } else if (layout == ValueLayout.JAVA_INT) {
+      !(ptr + offset).cast[Int] = value.asInstanceOf[Int]
+    } else if (layout == ValueLayout.JAVA_LONG) {
+      !(ptr + offset).cast[Long] = value.asInstanceOf[Long]
+    } else if (layout == ValueLayout.JAVA_FLOAT) {
+      !(ptr + offset).cast[Float] = value.asInstanceOf[Float]
+    } else if (layout == ValueLayout.JAVA_DOUBLE) {
+      !(ptr + offset).cast[Double] = value.asInstanceOf[Double]
+    } else {
+      throw new UnsupportedOperationException("Unsupported layout in set")
+    }
```

Comment:
This is quite hacky, instead of `Any` (and hoping those two `Any`s correspond to each other), you should implement all (or a subset of the) overrides, just as it is in Java.

https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/foreign/MemorySegment.html

![image](https://github.com/user-attachments/assets/478fb7a4-70df-4923-845a-46899e83ffb3)

![image](https://github.com/user-attachments/assets/7291af6d-a985-4250-a0a8-74ed6fc986a7)